### PR TITLE
Add Pack Iteration blog post

### PR DESCRIPTION
**Pull request by simanerush**
Add a blog post about pack iteration ([SE-0408](https://github.com/apple/swift-evolution/blob/main/proposals/0408-pack-iteration.md)) and how to use it, featuring a brief overview of parameter packs ([SE-0393](https://github.com/apple/swift-evolution/blob/main/proposals/0393-parameter-packs.md)).

This is my first blog post, so I will appreciate any feedback! 

### Motivation:

With the introduction of parameter packs in Swift 5.9, there was a demand for pack iteration support using a `for`-`in` loop syntax. [SE-0408](https://github.com/apple/swift-evolution/blob/main/proposals/0408-pack-iteration.md) enabled pack iteration for Swift 6.0, and it makes interacting with parameter packs way more straightforward for a Swift programmer. I wrote this blog post to introduce folks with parameter packs and show a new way of interacting with them using a `for`-`in` loop 🙂

### Modifications:

A new blog post about pack iteration.

### Result:

A new blog post will be added to [swift.org](https://www.swift.org). 

### Acknowledgements

Thank you @DougGregor for suggesting me to write this and reviewing the initial drafts.

### CODE REVIEW

File: _posts/2024-03-03-pack-iteration.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+published: true
+date: 2024-03-03 10:00:00
```
Comment (DougGregor) [inline _posts/2024-03-03-pack-iteration.md:4]: Let's remember to update this when we merge

File: _posts/2024-03-03-pack-iteration.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+title: Iterate Over Parameter Packs in Swift 6.0
+author: [simanerush]
+---
+
+Parameter packs, introduced in Swift 5.9, make it possible to write generics that abstract over the number of arguments, meaning you don’t need to have overloaded copies of the same generic function for one argument, two arguments, three arguments, and so on.
```
Comment (DougGregor) [inline _posts/2024-03-03-pack-iteration.md:9]:
```suggestion
Parameter packs, introduced in Swift 5.9, make it possible to write generics that abstract over the number of arguments. The eliminates the need to have overloaded copies of the same generic function for one argument, two arguments, three arguments, and so on.
```

Comment (dempseyatgithub) [reply _posts/2024-03-03-pack-iteration.md:9]: The suggestion should be "This eliminates the need", but I agree a new sentence works well here.

Comment (DougGregor) [reply _posts/2024-03-03-pack-iteration.md:9]: Fixed, thanks!

**Review by cthielen — APPROVED** [review]

**Comment by cthielen** [comment]
@swift-ci please test

**Pull request merged**